wasmer-api compilation fails with features = ["default-cranelift", "default-native"]

### The bug

I've added the following to `Cargo.toml`:

```toml
[dependencies.wasmer]
version = "1.0.0-alpha01.0"
default-features = false
features = ["default-cranelift", "default-native"]
```

running `cargo +nightly build` returns:

```
...
...

   Compiling wasmer v1.0.0-alpha01.0
error[E0308]: mismatched types
   --> /Users/yaronwittenstein/.cargo/registry/src/github.com-1ecc6299db9ec823/wasmer-1.0.0-alpha01.0/src/store.rs:104:55
    |
104 |                     wasmer_engine_native::Native::new(&config)
    |                                                       ^^^^^^^ types differ in mutability
    |
    = note: expected mutable reference `&mut dyn wasmer_compiler::CompilerConfig`
                       found reference `&impl CompilerConfig + Send + Sync`
```

--- a/lib/api/src/store.rs
+++ b/lib/api/src/store.rs
@@ -94,14 +94,14 @@ impl Default for Store {
             }
         }
 
-        #[allow(unreachable_code)]
-        fn get_engine(config: impl CompilerConfig + Send + Sync) -> impl Engine + Send + Sync {
+        #[allow(unreachable_code, unused_mut)]
+        fn get_engine(mut config: impl CompilerConfig + Send + Sync) -> impl Engine + Send + Sync {
             cfg_if::cfg_if! {
                 if #[cfg(feature = "default-jit")] {
                     wasmer_engine_jit::JIT::new(&config)
                         .engine()
                 } else if #[cfg(feature = "default-native")] {
-                    wasmer_engine_native::Native::new(&config)
+                    wasmer_engine_native::Native::new(&mut config)
                         .engine()
                 } else {
                     compile_error!("No default engine chosen")

